Dockerfile.testing

```diff
@@ -0,0 +1,82 @@
+FROM golang:1.12-stretch
+MAINTAINER TRON-US <support@tron.network>
+# Dockerfile.testing will build an image that contains the go-btfs source and binary.
+# It is quite large.  Its primary use case is to run the unit tests and test/sharness tests.
+# Use Dockerfile to run a btfs daemon instead
+ENV SRC_DIR /go-btfs
+
+# Download packages first so they can be cached.
+COPY go.mod go.sum $SRC_DIR/
+RUN cd $SRC_DIR \
+  && go mod download
+
+COPY . $SRC_DIR
+
+# Newer git submodule uses "absorbgitdirs" option by default which does not
+# include .git folder inside a submodule.
+# Use a build time variable $gitdir to specify the location of the actual .git folder.
+ARG gitdir=.git
+RUN test -d $SRC_DIR/.git \
+  || mv $SRC_DIR/$gitdir $SRC_DIR/.git
+
+# Install path
+RUN apt-get update && apt-get install -y patch
+
+# Build the thing.
+# Also: fix getting HEAD commit hash via git rev-parse.
+RUN cd $SRC_DIR \
+  && mkdir .git/objects \
+  && make build
+
+ENV SUEXEC_VERSION v0.2
+ENV TINI_VERSION v0.16.1
+RUN set -x \
+  && cd /tmp \
+  && git clone https://github.com/ncopa/su-exec.git \
+  && cd su-exec \
+  && git checkout -q $SUEXEC_VERSION \
+  && make \
+  && cd /tmp \
+  && wget -q -O tini https://github.com/krallin/tini/releases/download/$TINI_VERSION/tini \
+  && chmod +x tini
+
+# Get the TLS CA certificates, they're not provided by busybox.
+RUN apt-get update && apt-get install -y ca-certificates
+
+# Install FUSE
+RUN apt-get update && apt-get install -y fuse
+
+# Do this in the current container
+RUN cp /tmp/su-exec/su-exec /sbin/su-exec
```

Comment:
Why are we creating copies here?

Reply:
we could also do a mv instead of a copy. we need the su-exec command somewhere in the PATH.